sys/include/byteorder.h

```diff
@@ -682,6 +683,93 @@ static inline void byteorder_htobebufll(uint8_t *buf, uint64_t val)
 #endif
 }
 
+static inline uint16_t byteorder_lebuftohs(const uint8_t *buf)
+{
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+    /* little endian to little endian conversion is easy, but buffer might be
+     * unaligned */
+    return unaligned_get_u16(buf);
+#elif __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+    return (uint16_t)((buf[0] << 8) | (buf[1] << 0));
+#endif
+}
+
+static inline uint32_t byteorder_lebuftohl(const uint8_t *buf)
+{
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+    /* little endian to little endian conversion is easy, but buffer might be
+     * unaligned */
+    return unaligned_get_u32(buf);
+#elif __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+    return (((uint32_t) buf[0] << 24)
+          | ((uint32_t) buf[1] << 16)
+          | ((uint32_t) buf[2] << 8)
+          | ((uint32_t) buf[3] << 0));
+#endif
+}
+
+static inline uint64_t byteorder_lebuftohll(const uint8_t *buf)
+{
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+    /* little endian to little endian conversion is easy, but buffer might be
+     * unaligned */
+    return unaligned_get_u64(buf);
+#elif __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+    return (((uint64_t) buf[0] << 56)
+          | ((uint64_t) buf[1] << 48)
+          | ((uint64_t) buf[2] << 40)
+          | ((uint64_t) buf[3] << 32)
+          | ((uint64_t) buf[4] << 24)
+          | ((uint64_t) buf[5] << 16)
+          | ((uint64_t) buf[6] <<  8)
+          | ((uint64_t) buf[7] <<  0));
+#endif
+}
+
+static inline void byteorder_htolebufs(uint8_t *buf, uint16_t val)
+{
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
```

Comment:
even though memcpy might be slightly faster (which I doubt, it would be inlined and compiler optimized anyways), wouldn't it make sense to just re-use the big endian version? It should work in both cases, right?

Reply:
No sure I understand. Both have different semantics, thats the reason for having these functions added, right?!

Say I am on a little endian platform:
- calling `byteorder_htolebufs` will translate to `memcpy`
- calling `byteorder_htobebufs` will swap the bytes
So how do you imagin to put both into the same function?

> even though memcpy might be slightly faster 
But I don't want to use `memcpy` in my code directly, as this would break the code if it would be compiled for a big endian platform...

Reply:
ahh, now I think I got it... You mean re-using the swapping code snippets for both le/be functions, right?

Reply:
`byteorder_htolebufs()` is supposed to write val from host byte order to buf in little endian.

```
buf[0] = (uint8_t)(val >> 8);
buf[1] = (uint8_t)(val >> 0);
```

...does that, independent of the host byte order.
It would be nice to have only one implementation that we know is endian independent, and not ifdef to memcpy on one or the other.

Reply:
From https://commandcenter.blogspot.com/2012/04/byte-order-fallacy.html:

"If your code contains #ifdef BIG_ENDIAN or the equivalent, you need to unlearn about byte order."

Reply:
> wouldn't it make sense to just re-use the big endian version?

I meant, the big endian (manual byte write) else case in each function. :)